responses/__init__.py

```diff
@@ -1091,14 +1094,18 @@ def _on_request(
                 return response
         return response
 
-    def start(self) -> None:
-        if self._patcher:
-            # we must not override value of the _patcher if already applied
-            # this prevents issues when one decorated function is called from
-            # another decorated function
-            return
-
-        def unbound_on_send(
+    def unbound_on_send(
+        self,
+    ) -> """Callable[  # noqa: F821
+        [
+            Arg(HTTPAdapter, "adapter"),
+            Arg(PreparedRequest, "request"),
```

Comment:
@asottile-sentry can you please may be suggest here?

For some reason flake8 complains that
```
responses/__init__.py:1101:30: F821 undefined name 'adapter'
responses/__init__.py:1102:34: F821 undefined name 'request'
```

but `Arg` from mypy extensions takes just a name of the positional argument. So, what is not located by `flake8` ?

Reply:
use a `Protocol` -- `Arg` / etc. are deprecated mypy-specific types and will not work with other type checkers

Reply:
@asottile-sentry gotcha, thanks

fixed in the next commit